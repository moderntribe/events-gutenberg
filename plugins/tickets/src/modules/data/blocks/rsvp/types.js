/**
 * Internal dependencies
 */
import { PREFIX_TICKETS_STORE } from '@moderntribe/tickets/data/utils';

export const CREATE_RSVP = `${ PREFIX_TICKETS_STORE }/CREATE_RSVP`;
export const INITIALIZE_RSVP = `${ PREFIX_TICKETS_STORE }/INITIALIZE_RSVP`;
export const DELETE_RSVP = `${ PREFIX_TICKETS_STORE }/DELETE_RSVP`;

export const SET_RSVP_ID = `${ PREFIX_TICKETS_STORE }/SET_RSVP_ID`;
export const SET_RSVP_SETTINGS_OPEN = `${ PREFIX_TICKETS_STORE }/SET_RSVP_SETTINGS_OPEN`;
export const SET_RSVP_HAS_CHANGES = `${ PREFIX_TICKETS_STORE }/SET_RSVP_HAS_CHANGES`;
export const SET_RSVP_IS_LOADING = `${ PREFIX_TICKETS_STORE }/SET_RSVP_IS_LOADING`;
export const SET_RSVP_IS_SETTINGS_LOADING = `${ PREFIX_TICKETS_STORE }/SET_RSVP_IS_SETTINGS_LOADING`;
export const SET_RSVP_GOING_COUNT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_GOING_COUNT`;
export const SET_RSVP_NOT_GOING_COUNT = `${ PREFIX_TICKETS_STORE }/SET_RSVP_NOT_GOING_COUNT`;

export const SET_RSVP_DETAILS = `${ PREFIX_TICKETS_STORE }/SET_RSVP_DETAILS`;
export const SET_RSVP_TEMP_DETAILS = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_DETAILS`;
export const SET_RSVP_HEADER_IMAGE = `${ PREFIX_TICKETS_STORE }/SET_RSVP_HEADER_IMAGE`;

export const SET_RSVP_TITLE = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TITLE`;
export const SET_RSVP_DESCRIPTION = `${ PREFIX_TICKETS_STORE }/SET_RSVP_DESCRIPTION`;
export const SET_RSVP_CAPACITY = `${ PREFIX_TICKETS_STORE }/SET_RSVP_CAPACITY`;
export const SET_RSVP_NOT_GOING_RESPONSES = `${ PREFIX_TICKETS_STORE }/SET_RSVP_NOT_GOING_RESPONSES`;
export const SET_RSVP_START_DATE = `${ PREFIX_TICKETS_STORE }/SET_RSVP_START_DATE`;
export const SET_RSVP_START_DATE_OBJ = `${ PREFIX_TICKETS_STORE }/SET_RSVP_START_DATE_OBJ`;
export const SET_RSVP_END_DATE = `${ PREFIX_TICKETS_STORE }/SET_RSVP_END_DATE`;
export const SET_RSVP_END_DATE_OBJ = `${ PREFIX_TICKETS_STORE }/SET_RSVP_END_DATE_OBJ`;
export const SET_RSVP_START_TIME = `${ PREFIX_TICKETS_STORE }/SET_RSVP_START_TIME`;
export const SET_RSVP_END_TIME = `${ PREFIX_TICKETS_STORE }/SET_RSVP_END_TIME`;

export const SET_RSVP_TEMP_TITLE = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_TITLE`;
export const SET_RSVP_TEMP_DESCRIPTION = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_DESCRIPTION`;
export const SET_RSVP_TEMP_CAPACITY = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_CAPACITY`;
export const SET_RSVP_TEMP_NOT_GOING_RESPONSES = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_NOT_GOING_RESPONSES`;
export const SET_RSVP_TEMP_START_DATE = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_START_DATE`;
export const SET_RSVP_TEMP_START_DATE_OBJ = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_START_DATE_OBJ`;
export const SET_RSVP_TEMP_END_DATE = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_END_DATE`;
export const SET_RSVP_TEMP_END_DATE_OBJ = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_END_DATE_OBJ`;
export const SET_RSVP_TEMP_START_TIME = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_START_TIME`;
export const SET_RSVP_TEMP_END_TIME = `${ PREFIX_TICKETS_STORE }/SET_RSVP_TEMP_END_TIME`;
