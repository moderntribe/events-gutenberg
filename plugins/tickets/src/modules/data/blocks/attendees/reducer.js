/**
 * Internal dependencies
 */
import * as types from './types';

export const DEFAULT_STATE = {
	title: '',
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_ATTENDEES_TITLE:
			return {
				...state,
				title: action.payload.title,
			};
		default:
			return state;
	}
};
