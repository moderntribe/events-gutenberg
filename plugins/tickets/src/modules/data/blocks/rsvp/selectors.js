/**
 * External dependencies
 */
import { createSelector } from 'reselect';

export const getRSVPBlock = ( state ) => state.tickets.blocks.rsvp;

export const getRSVPId = createSelector(
	[ getRSVPBlock ],
	( rsvp ) => rsvp.id,
);

export const getRSVPCreated = createSelector(
	[ getRSVPBlock ],
	( rsvp ) => rsvp.created,
);

export const getRSVPSettingsOpen = createSelector(
	[ getRSVPBlock ],
	( rsvp ) => rsvp.settingsOpen,
);

export const getRSVPHasChanges = createSelector(
	[ getRSVPBlock ],
	( rsvp ) => rsvp.hasChanges,
);

export const getRSVPIsLoading = createSelector(
	[ getRSVPBlock ],
	( rsvp ) => rsvp.isLoading,
);

export const getRSVPIsSettingsLoading = createSelector(
	[ getRSVPBlock ],
	( rsvp ) => rsvp.isSettingsLoading,
);

export const getRSVPGoingCount = createSelector(
	[ getRSVPBlock ],
	( rsvp ) => rsvp.goingCount,
);

export const getRSVPNotGoingCount = createSelector(
	[ getRSVPBlock ],
	( rsvp ) => rsvp.notGoingCount,
);

/**
 * ------------------------------------------------------------
 * RSVP Details
 * ------------------------------------------------------------
 */
export const getRSVPDetails = createSelector(
	[ getRSVPBlock ],
	( rsvp ) => rsvp.details,
);

export const getRSVPTitle = createSelector(
	[ getRSVPDetails ],
	( details ) => details.title,
);

export const getRSVPDescription = createSelector(
	[ getRSVPDetails ],
	( details ) => details.description,
);

export const getRSVPCapacity = createSelector(
	[ getRSVPDetails ],
	( details ) => details.capacity,
);

export const getRSVPAvailable = createSelector(
	[ getRSVPCapacity, getRSVPGoingCount ],
	( capacity, goingCount ) => {
		if ( capacity === '' ) {
			return -1;
		};

		const total = parseInt( capacity, 10 ) || 0;
		const going = parseInt( goingCount, 10 ) || 0;
		/**
		 * Prevent to have negative values when subtracting the going amount from total amount, so it takes the max value
		 * of the substraction operation or zero if the operation is lower than zero it will return zero insted.
		 */
		return Math.max( total - going, 0 );
	}
)

export const getRSVPNotGoingResponses = createSelector(
	[ getRSVPDetails ],
	( details ) => details.notGoingResponses,
);

export const getRSVPStartDate = createSelector(
	[ getRSVPDetails ],
	( details ) => details.startDate,
);

export const getRSVPStartDateObj = createSelector(
	[ getRSVPDetails ],
	( details ) => details.startDateObj,
);

export const getRSVPStartTime = createSelector(
	[ getRSVPDetails ],
	( details ) => details.startTime,
);

export const getRSVPEndDate = createSelector(
	[ getRSVPDetails ],
	( details ) => details.endDate,
);

export const getRSVPEndDateObj = createSelector(
	[ getRSVPDetails ],
	( details ) => details.endDateObj,
);

export const getRSVPEndTime = createSelector(
	[ getRSVPDetails ],
	( details ) => details.endTime,
);

/**
 * ------------------------------------------------------------
 * RSVP Temp Details
 * ------------------------------------------------------------
 */
export const getRSVPTempDetails = createSelector(
	[ getRSVPBlock ],
	( rsvp ) => rsvp.tempDetails,
);

export const getRSVPTempTitle = createSelector(
	[ getRSVPTempDetails ],
	( tempDetails ) => tempDetails.title,
);

export const getRSVPTempDescription = createSelector(
	[ getRSVPTempDetails ],
	( tempDetails ) => tempDetails.description,
);

export const getRSVPTempCapacity = createSelector(
	[ getRSVPTempDetails ],
	( tempDetails ) => tempDetails.capacity,
);

export const getRSVPTempNotGoingResponses = createSelector(
	[ getRSVPTempDetails ],
	( tempDetails ) => tempDetails.notGoingResponses,
);

export const getRSVPTempStartDate = createSelector(
	[ getRSVPTempDetails ],
	( tempDetails ) => tempDetails.startDate,
);

export const getRSVPTempStartDateObj = createSelector(
	[ getRSVPTempDetails ],
	( tempDetails ) => tempDetails.startDateObj,
);

export const getRSVPTempStartTime = createSelector(
	[ getRSVPTempDetails ],
	( tempDetails ) => tempDetails.startTime,
);

export const getRSVPTempEndDate = createSelector(
	[ getRSVPTempDetails ],
	( tempDetails ) => tempDetails.endDate,
);

export const getRSVPTempEndDateObj = createSelector(
	[ getRSVPTempDetails ],
	( tempDetails ) => tempDetails.endDateObj,
);

export const getRSVPTempEndTime = createSelector(
	[ getRSVPTempDetails ],
	( tempDetails ) => tempDetails.endTime,
);

/**
 * ------------------------------------------------------------
 * RSVP Header Image
 * ------------------------------------------------------------
 */
export const getRSVPHeaderImage = createSelector(
	[ getRSVPBlock ],
	( rsvp ) => rsvp.headerImage,
);

export const getRSVPHeaderImageId = createSelector(
	[ getRSVPHeaderImage ],
	( headerImage ) => headerImage.id,
);

export const getRSVPHeaderImageSrc = createSelector(
	[ getRSVPHeaderImage ],
	( headerImage ) => headerImage.src,
);

export const getRSVPHeaderImageAlt = createSelector(
	[ getRSVPHeaderImage ],
	( headerImage ) => headerImage.alt,
);
