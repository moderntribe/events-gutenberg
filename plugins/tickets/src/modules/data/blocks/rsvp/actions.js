/**
 * Internal dependencies
 */
import { types } from '@moderntribe/tickets/data/blocks/rsvp';

export const createRSVP = () => ( {
	type: types.CREATE_RSVP,
} );

export const initializeRSVP = () => ( {
	type: types.INITIALIZE_RSVP,
} );

export const deleteRSVP = () => ( {
	type: types.DELETE_RSVP,
} );

export const setRSVPId = ( id ) => ( {
	type: types.SET_RSVP_ID,
	payload: {
		id,
	},
} );

export const setRSVPSettingsOpen = ( settingsOpen ) => ( {
	type: types.SET_RSVP_SETTINGS_OPEN,
	payload: {
		settingsOpen,
	},
} );

export const setRSVPHasChanges = ( hasChanges ) => ( {
	type: types.SET_RSVP_HAS_CHANGES,
	payload: {
		hasChanges,
	},
} );

export const setRSVPIsLoading = ( isLoading ) => ( {
	type: types.SET_RSVP_IS_LOADING,
	payload: {
		isLoading,
	},
} );

export const setRSVPIsSettingsLoading = ( isSettingsLoading ) => ( {
	type: types.SET_RSVP_IS_SETTINGS_LOADING,
	payload: {
		isSettingsLoading,
	},
} );

export const setRSVPGoingCount = ( goingCount ) => ( {
	type: types.SET_RSVP_GOING_COUNT,
	payload: {
		goingCount,
	},
} );

export const setRSVPNotGoingCount = ( notGoingCount ) => ( {
	type: types.SET_RSVP_NOT_GOING_COUNT,
	payload: {
		notGoingCount,
	},
} );

export const setRSVPTitle = ( title ) => ( {
	type: types.SET_RSVP_TITLE,
	payload: {
		title,
	},
} );

export const setRSVPDescription = ( description ) => ( {
	type: types.SET_RSVP_DESCRIPTION,
	payload: {
		description,
	},
} );

export const setRSVPCapacity = ( capacity ) => ( {
	type: types.SET_RSVP_CAPACITY,
	payload: {
		capacity,
	},
} );

export const setRSVPNotGoingResponses = ( notGoingResponses ) => ( {
	type: types.SET_RSVP_NOT_GOING_RESPONSES,
	payload: {
		notGoingResponses,
	},
} );

export const setRSVPStartDate = ( startDate ) => ( {
	type: types.SET_RSVP_START_DATE,
	payload: {
		startDate,
	},
} );

export const setRSVPStartDateObj = ( startDateObj ) => ( {
	type: types.SET_RSVP_START_DATE_OBJ,
	payload: {
		startDateObj,
	},
} );

export const setRSVPEndDate = ( endDate ) => ( {
	type: types.SET_RSVP_END_DATE,
	payload: {
		endDate,
	},
} );

export const setRSVPEndDateObj = ( endDateObj ) => ( {
	type: types.SET_RSVP_END_DATE_OBJ,
	payload: {
		endDateObj,
	},
} );

export const setRSVPStartTime = ( startTime ) => ( {
	type: types.SET_RSVP_START_TIME,
	payload: {
		startTime,
	},
} );

export const setRSVPEndTime = ( endTime ) => ( {
	type: types.SET_RSVP_END_TIME,
	payload: {
		endTime,
	},
} );

export const setRSVPTempTitle = ( title ) => ( {
	type: types.SET_RSVP_TEMP_TITLE,
	payload: {
		title,
	},
} );

export const setRSVPTempDescription = ( description ) => ( {
	type: types.SET_RSVP_TEMP_DESCRIPTION,
	payload: {
		description,
	},
} );

export const setRSVPTempCapacity = ( capacity ) => ( {
	type: types.SET_RSVP_TEMP_CAPACITY,
	payload: {
		capacity,
	},
} );

export const setRSVPTempNotGoingResponses = ( notGoingResponses ) => ( {
	type: types.SET_RSVP_TEMP_NOT_GOING_RESPONSES,
	payload: {
		notGoingResponses,
	},
} );

export const setRSVPTempStartDate = ( startDate ) => ( {
	type: types.SET_RSVP_TEMP_START_DATE,
	payload: {
		startDate,
	},
} );

export const setRSVPTempStartDateObj = ( startDateObj ) => ( {
	type: types.SET_RSVP_TEMP_START_DATE_OBJ,
	payload: {
		startDateObj,
	},
} );

export const setRSVPTempEndDate = ( endDate ) => ( {
	type: types.SET_RSVP_TEMP_END_DATE,
	payload: {
		endDate,
	},
} );

export const setRSVPTempEndDateObj = ( endDateObj ) => ( {
	type: types.SET_RSVP_TEMP_END_DATE_OBJ,
	payload: {
		endDateObj,
	},
} );

export const setRSVPTempStartTime = ( startTime ) => ( {
	type: types.SET_RSVP_TEMP_START_TIME,
	payload: {
		startTime,
	},
} );

export const setRSVPTempEndTime = ( endTime ) => ( {
	type: types.SET_RSVP_TEMP_END_TIME,
	payload: {
		endTime,
	},
} );

export const setRSVPHeaderImage = ( payload ) => ( {
	type: types.SET_RSVP_HEADER_IMAGE,
	payload,
} );

export const setRSVPDetails = ( payload ) => ( {
	type: types.SET_RSVP_DETAILS,
	payload,
} );

export const setRSVPTempDetails = ( payload ) => ( {
	type: types.SET_RSVP_TEMP_DETAILS,
	payload,
} );
