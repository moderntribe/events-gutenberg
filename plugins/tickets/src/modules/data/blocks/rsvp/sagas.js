/**
 * External Dependencies
 */
import { put, call, all, select, takeEvery } from 'redux-saga/effects';

/**
 * Internal dependencies
 */
import * as types from './types';
import * as actions from './actions';
import { getStart, getEnd } from '@moderntribe/events/data/blocks/datetime/selectors';
import { toMoment, toDate, toTime24Hr } from '@moderntribe/common/utils/moment';

export function* setRSVPDetails( action ) {
	const {
		title,
		description,
		capacity,
		notGoingResponses,
		startDate,
		startDateObj,
		startTime,
		endDate,
		endDateObj,
		endTime,
	} = action.payload;
	yield all( [
		put( actions.setRSVPTitle( title ) ),
		put( actions.setRSVPDescription( description ) ),
		put( actions.setRSVPCapacity( capacity ) ),
		put( actions.setRSVPNotGoingResponses( notGoingResponses ) ),
		put( actions.setRSVPStartDate( startDate ) ),
		put( actions.setRSVPStartDateObj( startDateObj ) ),
		put( actions.setRSVPStartTime( startTime ) ),
		put( actions.setRSVPEndDate( endDate ) ),
		put( actions.setRSVPEndDateObj( endDateObj ) ),
		put( actions.setRSVPEndTime( endTime ) ),
	] );
}

export function* setRSVPTempDetails( action ) {
	const {
		tempTitle,
		tempDescription,
		tempCapacity,
		tempNotGoingResponses,
		tempStartDate,
		tempStartDateObj,
		tempStartTime,
		tempEndDate,
		tempEndDateObj,
		tempEndTime,
	} = action.payload;
	yield all( [
		put( actions.setRSVPTempTitle( tempTitle ) ),
		put( actions.setRSVPTempDescription( tempDescription ) ),
		put( actions.setRSVPTempCapacity( tempCapacity ) ),
		put( actions.setRSVPTempNotGoingResponses( tempNotGoingResponses ) ),
		put( actions.setRSVPTempStartDate( tempStartDate ) ),
		put( actions.setRSVPTempStartDateObj( tempStartDateObj ) ),
		put( actions.setRSVPTempStartTime( tempStartTime ) ),
		put( actions.setRSVPTempEndDate( tempEndDate ) ),
		put( actions.setRSVPTempEndDateObj( tempEndDateObj ) ),
		put( actions.setRSVPTempEndTime( tempEndTime ) ),
	] );
}

export function* initializeRSVP() {
	const start = yield select( getStart );
	const end = yield select( getEnd );

	const startMoment = yield call( toMoment, start );
	const endMoment = yield call( toMoment, end );

	const startDate = yield call( toDate, startMoment );
	const endDate = yield call( toDate, endMoment );

	const startTime = yield call( toTime24Hr, startMoment );
	const endTime = yield call( toTime24Hr, endMoment );

	const startDateObj = new Date( startDate );
	const endDateObj = new Date( endDate );

	yield all( [
		put( actions.setRSVPTempStartDate( startDate ) ),
		put( actions.setRSVPTempStartDateObj( startDateObj ) ),
		put( actions.setRSVPTempStartTime( startTime ) ),
		put( actions.setRSVPTempEndDate( endDate ) ),
		put( actions.setRSVPTempEndDateObj( endDateObj ) ),
		put( actions.setRSVPTempEndTime( endTime ) ),
	] );
}

export default function* watchers() {
	yield takeEvery( types.SET_RSVP_DETAILS, setRSVPDetails );
	yield takeEvery( types.SET_RSVP_TEMP_DETAILS, setRSVPTempDetails );
	yield takeEvery( types.INITIALIZE_RSVP, initializeRSVP );
}
