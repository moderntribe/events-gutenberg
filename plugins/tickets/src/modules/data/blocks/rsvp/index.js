/**
 * Internal dependencies
 */
import * as types from './types';
import * as actions from './actions';
import * as selectors from './selectors';
import reducer from './reducer';
import sagas from './sagas';

export default reducer;
export { types, actions, sagas, selectors };
