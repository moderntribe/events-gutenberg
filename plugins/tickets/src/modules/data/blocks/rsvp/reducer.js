/**
 * External dependencies
 */
import moment from 'moment/moment';

/**
 * Internal dependencies
 */
import { types } from '@moderntribe/tickets/data/blocks/rsvp';
import { moment as momentUtil } from '@moderntribe/common/utils';

export const DEFAULT_STATE = {
	title: '',
	description: '',
	capacity: '',
	notGoingResponses: false,
	startDate: momentUtil.toDate( moment() ),
	endDate: momentUtil.toDate( moment() ),
	startTime: momentUtil.toTime24Hr( moment() ),
	endTime: momentUtil.toTime24Hr( moment() ),
	tempTitle: '',
	tempDescription: '',
	tempCapacity: '',
	tempNotGoingResponses: false,
	tempStartDate: momentUtil.toDate( moment() ),
	tempStartDateObj: new Date( momentUtil.toDate( moment() ) ),
	tempEndDate: momentUtil.toDate( moment() ),
	tempEndDateObj: new Date( momentUtil.toDate( moment() ) ),
	tempStartTime: momentUtil.toTime24Hr( moment() ),
	tempEndTime: momentUtil.toTime24Hr( moment() ),
	created: false,
	settingsOpen: false,
	hasChanges: false,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.CREATE_RSVP:
			return {
				...state,
				created: true,
			};
		case types.DELETE_RSVP:
			return DEFAULT_STATE;
		case types.SET_RSVP_TITLE:
			return {
				...state,
				title: action.payload.title,
			};
		case types.SET_RSVP_DESCRIPTION:
			return {
				...state,
				description: action.payload.description,
			};
		case types.SET_RSVP_CAPACITY:
			return {
				...state,
				capacity: action.payload.capacity,
			};
		case types.SET_RSVP_NOT_GOING_RESPONSES:
			return {
				...state,
				notGoingResponses: action.payload.notGoingResponses,
			};
		case types.SET_RSVP_START_DATE:
			return {
				...state,
				startDate: action.payload.startDate,
			};
		case types.SET_RSVP_END_DATE:
			return {
				...state,
				endDate: action.payload.endDate,
			};
		case types.SET_RSVP_START_TIME:
			return {
				...state,
				startTime: action.payload.startTime,
			};
		case types.SET_RSVP_END_TIME:
			return {
				...state,
				endTime: action.payload.endTime,
			};
		case types.SET_RSVP_TEMP_TITLE:
			return {
				...state,
				tempTitle: action.payload.tempTitle,
			};
		case types.SET_RSVP_TEMP_DESCRIPTION:
			return {
				...state,
				tempDescription: action.payload.tempDescription,
			};
		case types.SET_RSVP_TEMP_CAPACITY:
			return {
				...state,
				tempCapacity: action.payload.tempCapacity,
			};
		case types.SET_RSVP_TEMP_NOT_GOING_RESPONSES:
			return {
				...state,
				tempNotGoingResponses: action.payload.tempNotGoingResponses,
			};
		case types.SET_RSVP_TEMP_START_DATE:
			return {
				...state,
				tempStartDate: action.payload.tempStartDate,
			};
		case types.SET_RSVP_TEMP_START_DATE_OBJ:
			return {
				...state,
				tempStartDateObj: action.payload.tempStartDateObj,
			};
		case types.SET_RSVP_TEMP_END_DATE:
			return {
				...state,
				tempEndDate: action.payload.tempEndDate,
			};
		case types.SET_RSVP_TEMP_END_DATE_OBJ:
			return {
				...state,
				tempEndDateObj: action.payload.tempEndDateObj,
			};
		case types.SET_RSVP_TEMP_START_TIME:
			return {
				...state,
				tempStartTime: action.payload.tempStartTime,
			};
		case types.SET_RSVP_TEMP_END_TIME:
			return {
				...state,
				tempEndTime: action.payload.tempEndTime,
			};
		case types.SET_RSVP_SETTINGS_OPEN:
			return {
				...state,
				settingsOpen: action.payload.settingsOpen,
			};
		case types.SET_RSVP_HAS_CHANGES:
			return {
				...state,
				hasChanges: action.payload.hasChanges,
			};
		default:
			return state;
	}
};
