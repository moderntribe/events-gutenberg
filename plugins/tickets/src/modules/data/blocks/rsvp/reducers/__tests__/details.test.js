/**
 * Internal dependencies
 */
import { actions } from '@moderntribe/tickets/data/blocks/rsvp';
import reducer, { DEFAULT_STATE } from '@moderntribe/tickets/data/blocks/rsvp/reducers/details';

jest.mock( 'moment', () => () => {
	const moment = require.requireActual( 'moment' );
	return moment( 'July 19, 2018 7:30 pm', 'MMMM D, Y h:mm a' );
} );

describe( 'Details reducer', () => {
	it( 'should set the default state', () => {
		expect( reducer( undefined, {} ) ).toEqual( DEFAULT_STATE );
	} );

	it( 'should set the title', () => {
		expect( reducer( DEFAULT_STATE, actions.setRSVPTitle( 'new title' ) ) ).toMatchSnapshot();
	} );

	it( 'should set the description', () => {
		expect( reducer( DEFAULT_STATE, actions.setRSVPDescription( 'new description' ) ) ).toMatchSnapshot();
	} );

	it( 'should set the capacity', () => {
		expect( reducer( DEFAULT_STATE, actions.setRSVPCapacity( 20 ) ) ).toMatchSnapshot();
	} );

	it( 'should set the not going responses', () => {
		expect( reducer( DEFAULT_STATE, actions.setRSVPNotGoingResponses( true ) ) ).toMatchSnapshot();
	} );

	it( 'should set the start date', () => {
		expect( reducer( DEFAULT_STATE, actions.setRSVPStartDate( 'January 1, 2018' ) ) ).toMatchSnapshot();
	} );

	it( 'should set the start date object', () => {
		expect( reducer(
			DEFAULT_STATE,
			actions.setRSVPStartDateObj( new Date( 'January 1, 2018' ) ),
		) ).toMatchSnapshot();
	} );

	it( 'should set the end date', () => {
		expect( reducer( DEFAULT_STATE, actions.setRSVPEndDate( 'January 1, 2018' ) ) ).toMatchSnapshot();
	} );

	it( 'should set the end date object', () => {
		expect( reducer(
			DEFAULT_STATE,
			actions.setRSVPEndDateObj( new Date( 'January 1, 2018' ) ),
		) ).toMatchSnapshot();
	} );

	it( 'should set the start time', () => {
		expect( reducer( DEFAULT_STATE, actions.setRSVPStartTime( '13:45' ) ) ).toMatchSnapshot();
	} );

	it( 'should set the end time', () => {
		expect( reducer( DEFAULT_STATE, actions.setRSVPEndTime( '13:45' ) ) ).toMatchSnapshot();
	} );
} );
