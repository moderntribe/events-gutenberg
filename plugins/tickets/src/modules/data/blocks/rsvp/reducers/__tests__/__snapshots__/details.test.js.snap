// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details reducer should set the capacity 1`] = `
Object {
  "capacity": 20,
  "description": "",
  "endDate": "July 19, 2018",
  "endDateObj": 2018-07-19T00:00:00.000Z,
  "endTime": "19:30",
  "notGoingResponses": false,
  "startDate": "July 19, 2018",
  "startDateObj": 2018-07-19T00:00:00.000Z,
  "startTime": "19:30",
  "title": "",
}
`;

exports[`Details reducer should set the description 1`] = `
Object {
  "capacity": "",
  "description": "new description",
  "endDate": "July 19, 2018",
  "endDateObj": 2018-07-19T00:00:00.000Z,
  "endTime": "19:30",
  "notGoingResponses": false,
  "startDate": "July 19, 2018",
  "startDateObj": 2018-07-19T00:00:00.000Z,
  "startTime": "19:30",
  "title": "",
}
`;

exports[`Details reducer should set the end date 1`] = `
Object {
  "capacity": "",
  "description": "",
  "endDate": "January 1, 2018",
  "endDateObj": 2018-07-19T00:00:00.000Z,
  "endTime": "19:30",
  "notGoingResponses": false,
  "startDate": "July 19, 2018",
  "startDateObj": 2018-07-19T00:00:00.000Z,
  "startTime": "19:30",
  "title": "",
}
`;

exports[`Details reducer should set the end date object 1`] = `
Object {
  "capacity": "",
  "description": "",
  "endDate": "July 19, 2018",
  "endDateObj": 2018-01-01T00:00:00.000Z,
  "endTime": "19:30",
  "notGoingResponses": false,
  "startDate": "July 19, 2018",
  "startDateObj": 2018-07-19T00:00:00.000Z,
  "startTime": "19:30",
  "title": "",
}
`;

exports[`Details reducer should set the end time 1`] = `
Object {
  "capacity": "",
  "description": "",
  "endDate": "July 19, 2018",
  "endDateObj": 2018-07-19T00:00:00.000Z,
  "endTime": "13:45",
  "notGoingResponses": false,
  "startDate": "July 19, 2018",
  "startDateObj": 2018-07-19T00:00:00.000Z,
  "startTime": "19:30",
  "title": "",
}
`;

exports[`Details reducer should set the not going responses 1`] = `
Object {
  "capacity": "",
  "description": "",
  "endDate": "July 19, 2018",
  "endDateObj": 2018-07-19T00:00:00.000Z,
  "endTime": "19:30",
  "notGoingResponses": true,
  "startDate": "July 19, 2018",
  "startDateObj": 2018-07-19T00:00:00.000Z,
  "startTime": "19:30",
  "title": "",
}
`;

exports[`Details reducer should set the start date 1`] = `
Object {
  "capacity": "",
  "description": "",
  "endDate": "July 19, 2018",
  "endDateObj": 2018-07-19T00:00:00.000Z,
  "endTime": "19:30",
  "notGoingResponses": false,
  "startDate": "January 1, 2018",
  "startDateObj": 2018-07-19T00:00:00.000Z,
  "startTime": "19:30",
  "title": "",
}
`;

exports[`Details reducer should set the start date object 1`] = `
Object {
  "capacity": "",
  "description": "",
  "endDate": "July 19, 2018",
  "endDateObj": 2018-07-19T00:00:00.000Z,
  "endTime": "19:30",
  "notGoingResponses": false,
  "startDate": "July 19, 2018",
  "startDateObj": 2018-01-01T00:00:00.000Z,
  "startTime": "19:30",
  "title": "",
}
`;

exports[`Details reducer should set the start time 1`] = `
Object {
  "capacity": "",
  "description": "",
  "endDate": "July 19, 2018",
  "endDateObj": 2018-07-19T00:00:00.000Z,
  "endTime": "19:30",
  "notGoingResponses": false,
  "startDate": "July 19, 2018",
  "startDateObj": 2018-07-19T00:00:00.000Z,
  "startTime": "13:45",
  "title": "",
}
`;

exports[`Details reducer should set the title 1`] = `
Object {
  "capacity": "",
  "description": "",
  "endDate": "July 19, 2018",
  "endDateObj": 2018-07-19T00:00:00.000Z,
  "endTime": "19:30",
  "notGoingResponses": false,
  "startDate": "July 19, 2018",
  "startDateObj": 2018-07-19T00:00:00.000Z,
  "startTime": "19:30",
  "title": "new title",
}
`;
